src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -354,25 +187,19 @@ export class DataView implements DataViewBase {
     return dataViewSpec;
   }
 
-  /**
-   * Get the source filtering configuration for that index.
-   */
-  getSourceFiltering() {
-    return {
-      excludes: (this.sourceFilters && this.sourceFilters.map((filter) => filter.value)) || [],
-    };
-  }
-
   /**
    * Removes scripted field from field list.
    * @param fieldName name of scripted field to remove
    * @deprecated use runtime field instead
    */
 
   removeScriptedField(fieldName: string) {
+    this.deleteScriptedFieldInternal(fieldName);
     const field = this.fields.getByName(fieldName);
-    if (field) {
+    if (field && field.scripted) {
       this.fields.remove(field);
+    } else {
+      throw new Error(`Scripted field ${fieldName} does not exist in data view ${this.getName()}`);
```

Comment:
Is it safe to start throwing here now? It didn't fail any tests, but could it be an issue somewhere without coverage?

Reply:
Its only used by REST endpoints and data view management ui, both places where throwing on a missing field is desired.